test/server/cards/01_SOR/units/GuardianOfTheWhills.spec.ts

```diff
@@ -0,0 +1,85 @@
+describe('Guardian of the Whills', function () {
+    integration(function (contextRef) {
+        describe('Guardian of the Whills\' ability', function () {
+            it('should decrease the cost of the first upgrade played on it by 1 resource, once per round', function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['resilient', 'devotion', 'electrostaff', 'foundling', 'entrenched', 'jedi-lightsaber'],
+                        groundArena: ['guardian-of-the-whills'],
+                        leader: 'chewbacca#walking-carpet', // vigilance aspect
+                    },
+                    player2: {
+                        hand: ['waylay'],
+                        spaceArena: ['system-patrol-craft'],
+                        leader: 'chewbacca#walking-carpet', // vigilance aspect
+                    }
+                });
+
+                const { context } = contextRef;
+
+                // The first upgrade put on the whills should be 1 cheaper
+                context.player1.clickCard(context.resilient);
+                expect(context.player1).toBeAbleToSelectExactly([context.guardianOfTheWhills, context.systemPatrolCraft]);
+                context.player1.clickCard(context.guardianOfTheWhills);
+                expect(context.guardianOfTheWhills.upgrades).toContain(context.resilient);
+                expect(context.player1.exhaustedResourceCount).toBe(0);
```

Comment:
Whenever we're testing a numerical increase / decrease I always want to use a higher base number to confirm the exact amount. In this case, can we use an upgrade that costs > 1?

Reply:
yes, fixed